language_service/src/completion.rs

```diff
@@ -432,3 +459,95 @@ impl Completion {
         }
     }
 }
+
+fn get_qasm_completions(
+    compilation: &Compilation,
+    _source_name: &str,
+    _position: Position,
+    _source_name_relative: &str,
+    source_contents: &str,
+    cursor_offset: u32,
+) -> CompletionList {
+    let expected_words_at_cursor =
+        qsc::qasm::completion::possible_words_at_offset_in_source(source_contents, cursor_offset);
+
+    // Now that we have the information from the parser about what kinds of
+    // words are expected, gather the actual words (identifiers, keywords, etc) for each kind.
+
+    // Keywords and other hardcoded words
+    let hardcoded_completions = collect_hardcoded_words_qasm(expected_words_at_cursor);
+
+    // The tricky bit: locals, names we need to gather from the compilation.
+    let name_completions = collect_names_qasm(expected_words_at_cursor, cursor_offset, compilation);
+
+    // We have all the data, put everything into a completion list.
+    into_completion_list(once(hardcoded_completions).chain(name_completions))
+}
+
+#[allow(clippy::items_after_statements)]
+fn collect_hardcoded_words_qasm(
```

Comment:
Has this been updated for QASM? still seems very Q# specific.

Reply:
What part seems Q# specific in hardcoded words?